sesman/sessionrecord.c

```diff
@@ -0,0 +1,131 @@
+/**
+ * xrdp: A Remote Desktop Protocol server.
+ *
+ * Copyright (C) Emmanuel Blindauer 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ * @file sessionrecord.c
+ * @brief utmp/wtmp handling code
+ * Idea: Only implement actual utmp, i.e. utmpx for 99%.
+ *       See http://80386.nl/unix/utmpx/
+ */
+
+#if defined(HAVE_CONFIG_H)
+#include <config_ac.h>
+#endif
+
+#include <paths.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "os_calls.h"
+#include "sessionrecord.h"
+
+/*
+ * Prepare the utmp/ struct and write it.
+ * this can handle login and logout at once with the 'state' parameter
+ */
+
+int
+add_xtmp_entry(int pid, const char *line, const char *user, const char *rhostname, short state)
```

Comment:
you never pass `state` argument when calling this function.
also, its unused here.

 consider removing it.

Reply:
They are used in ``utmp_login`` and ``utmp_logout``, ``state`` is set to  ``USER_PROCESS`` or ``DEAD_PROCESS``

Reply:
oh, right. my fault.
so just make it `const short`